### ISSUE TITLE
Indentation expected after templated code

### ISSUE THREAD

**Issue by tomslade1** [sqlfluff/sqlfluff#3455]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Following up [this](https://sqlfluff.slack.com/archives/C01GX4L213J/p1655150753464719) slack thread. 

We have some simple dbt macros in our project; that are triggering rule L003 violations immediately after the templated code; flagging that the indentations are out of alignment and an indentation is expected but not there.

We've tried altering the indents:
1.  In the macro
2. Around the macro 
3. By using whitespace control to strip out spaces/indents/lines from the template

But each time get a L003 error 🤯 

The macro which most frequently generates this error for us is:
```
{% macro base_limits(type, field) %}
{% set lookbacks = {'default': 14, 'receipt_journey': 1, 'segment': 3, 'calendars': 7, 'invoices': 3, 'salesforce': 3} %}
{% if target.name == 'dev' and not var('skip_limits', False) and not is_incremental() %}
    where {{ field }} > current_date - {{ lookbacks[type] }}
{% elif target.name == 'dev' and not var('skip_limits', False) and is_incremental() %}
    and {{ field }} > current_date - {{ lookbacks[type] }}
{% endif %}
{% endmacro %}
```

An example of usage is:
```
with subscriptions as (

    select *
    from {{ source('prepare', 'subscriptions') }}
{{ base_limits('default', 'created_at') }}

)

select *
from subscriptions
```
This produces:
`L:   5 | P:   1 | L003 | Expected 1 indentations, found 0 [compared to line 04]`

If we tweak the model to add an indent; it's not fixed and just yields the error one indent along:
```
with subscriptions as (

    select *
    from {{ source('prepare', 'subscriptions') }}
    {{ base_limits('default', 'created_at') }}

)

select *
from subscriptions
```
Linting failure:
`L:   5 | P:   5 | L003 | Expected 1 indentations, found 0 [compared to line 04]`

### Expected Behaviour

The indent failure is fixed, upon adding an indent.

### Observed Behaviour

Error moves upon adding the indent.

### How to reproduce

Example `.sql` file is detailed above. Macro stored in /macros path of dbt directory under `base_limits.sql`.

### Dialect

snowflake

### Version

sqlfluff, version 0.13.0

Python 3.9.13

dbt 1.0.7
dbt-snowflake plugin 1.0.1

### Configuration

.sqlfluff:
```
[sqlfluff]
templater = dbt
dialect = snowflake
capitalisation_policy = lower
verbose = 2

# L016 - Line is too long
# L029 - Keywords should not be used as identifiers
# L034 - Use wildcards then simple targets before calculations and aggregates in select statements
# L035 - Do not specify “else null” in a case when statement (redundant)
# L044 - Query produces an unknown number of result columns

exclude_rules = L016,
                L029,
                L034,
                L035,
                L044

[sqlfluff:templater:jinja]
load_macros_from_path = macros

[sqlfluff:rules]
capitalisation_policy = lower

[sqlfluff:rules:L029]
quoted_identifiers_policy = aliases
```

.sqlfluffignore:
`macros/`

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tomslade1** [comment]
FYI @nezd & @ivanovyordan

**Comment by barrywhart** [comment]
This was fixed in PR #3278, which was released in SQLFluff 0.13.2. Please update to that version.
